`test_autocomplete[_with_types]` fail with `textual` 6.5.0

When using the latest textual release ([6.5.0](https://github.com/Textualize/textual/releases/tag/v6.5.0)), the following two tests started failing:

```
=========================== short test summary info ============================
FAILED tests/functional_tests/test_autocomplete.py::test_autocomplete - AssertionError: assert None == 'word'
FAILED tests/functional_tests/test_autocomplete.py::test_autocomplete_with_types - AssertionError: assert None == 'word'
======================== 2 failed, 103 passed in 46.64s ========================
```

Logs:
```
______________________________ test_autocomplete _______________________________

app = TextEditorApp(title='TextEditorApp', classes={'-dark-mode'}, pseudo_classes={'dark', 'focus'})
word_completer = <function word_completer.<locals>._completer at 0x7ffff4dc80e0>

    @pytest.mark.asyncio
    async def test_autocomplete(
        app: App, word_completer: Callable[[str], list[tuple[str, str]]]
    ) -> None:
        messages: list[Message] = []
        async with app.run_test(message_hook=messages.append) as pilot:
            ta = app.query_one("#ta", expect_type=TextEditor)
            ta.word_completer = word_completer
            ta.focus()
            while ta.word_completer is None:
                await pilot.pause()

            start_time = monotonic()
            await pilot.press("s")
            while ta.completion_list.is_open is False:
                if monotonic() - start_time > 10:
                    print("MESSAGES:")
                    print("\n".join([str(m) for m in messages]))
                    break
                await pilot.pause()
            assert ta.text_input
>           assert ta.text_input.completer_active == "word"
E           AssertionError: assert None == 'word'
E            +  where None = TextAreaPlus().completer_active
E            +    where TextAreaPlus() = TextEditor(id='ta').text_input

tests/functional_tests/test_autocomplete.py:84: AssertionError
```

Context: [updating `textual` in nixpkgs](https://github.com/NixOS/nixpkgs/pull/457519).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+- **Breaking:** drops support for Textual &lt; 8.2.8.
+- Fixes a bug where focusing the `TextEditor` while it already had focus would dismiss an open autocomplete list. `TextEditor.focus()` now focuses the inner `TextArea` directly, instead of taking focus and forwarding it.
+- Adds <kbd>ctrl+backspace</kbd> and <kbd>alt+backspace</kbd> (delete word left) and <kbd>alt+delete</kbd> (delete word right), using Textual's own actions.
+- Adds <kbd>cmd</kbd> aliases for cut, copy, paste, undo, and redo, which Textual reports on terminals that support the Kitty keyboard protocol.
+- Copying now also writes to Textual's clipboard (which emits OSC 52), so copy works over ssh and in terminals where pyperclip has no backend.
+
 ## [0.17.3] - 2026-08-03
 
 - Double-, triple-, and quadruple-click selection now uses Textual's `Click.chain`, instead of tracking consecutive clicks with a timer ([harlequin/#708](https://github.com/tconbeer/harlequin/issues/708)).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,7 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
-- **Breaking:** drops support for Textual &lt; 8.2.8.
+- **Breaking:** drops support for Textual &lt; 7.3.
 - Fixes a bug where focusing the `TextEditor` while it already had focus would dismiss an open autocomplete list. `TextEditor.focus()` now focuses the inner `TextArea` directly, instead of taking focus and forwarding it.
 - Adds <kbd>ctrl+backspace</kbd> and <kbd>alt+backspace</kbd> (delete word left) and <kbd>alt+delete</kbd> (delete word right), using Textual's own actions.
 - Adds <kbd>cmd</kbd> aliases for cut, copy, paste, undo, and redo, which Textual reports on terminals that support the Kitty keyboard protocol.

--- a/README.md
+++ b/README.md
@@ -20,7 +20,8 @@ Full-featured text editor experience with VS-Code-like bindings, in your Textual
 - Syntax highlighting and support for Pygments themes.
 - Move cursor and scroll with mouse or keys (including <kbd>ctrl+arrow</kbd>, <kbd>PgUp/Dn</kbd>,  <kbd>ctrl+Home/End</kbd>).
 - Open (<kbd>ctrl+o</kbd>) and save (<kbd>ctrl+s</kbd>) files.
-- Cut (<kbd>ctrl+x</kbd>), copy (<kbd>ctrl+c</kbd>), paste (<kbd>ctrl+u/v</kbd>), optionally using the system clipboard.
+- Cut (<kbd>ctrl+x</kbd>), copy (<kbd>ctrl+c</kbd>), paste (<kbd>ctrl+u/v</kbd>), optionally using the system clipboard. On terminals that support the Kitty keyboard protocol, the <kbd>cmd</kbd> equivalents work too.
+- Delete a word with <kbd>ctrl+backspace</kbd>, <kbd>alt+backspace</kbd>, or <kbd>alt+delete</kbd>.
 - Comment selections with <kbd>ctrl+/</kbd>.
 - Indent and dedent (optionally for a multiline selection) to tab stops with <kbd>Tab</kbd> and <kbd>shift+Tab</kbd>.
 - Automatic completions of quotes and brackets.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,7 @@ authors = [
 license = "MIT"
 readme = "README.md"
 dependencies = [
-    "textual[syntax]>=6.4.0,<7",
+    "textual[syntax]>=8.2.8,<9",
     "pyperclip>=1.9.0,<2",
 ]
 
@@ -18,7 +18,7 @@ dependencies = [
 [dependency-groups]
 dev = [
     "pre-commit>=3.3,<5",
-    "textual==6.4.0",
+    "textual==8.2.8",
     "textual-dev==1.8.0",
     "pyinstrument>=5,<6",
 ]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,7 @@ authors = [
 license = "MIT"
 readme = "README.md"
 dependencies = [
-    "textual[syntax]>=8.2.8,<9",
+    "textual[syntax]>=7.3.0,<9",
     "pyperclip>=1.9.0,<2",
 ]
 

--- a/src/textual_textarea/text_editor.py
+++ b/src/textual_textarea/text_editor.py
@@ -1116,7 +1116,21 @@ class TextEditor(Widget, can_focus=True, can_focus_children=False):
         # self.text_input exists so watch_theme can do its thing.
         self.theme = self._theme
 
+    def focus(self, scroll_visible: bool = True) -> "TextEditor":
+        """
+        Focus the inner TextArea directly. Taking focus here first and then
+        forwarding it in on_focus would blur the TextArea on every call, and a
+        blur dismisses an open completion list.
+        """
+        if self.text_input is None:
+            super().focus(scroll_visible)
+        else:
+            self.text_input.focus(scroll_visible)
+        return self
+
     def on_focus(self) -> None:
+        # focus can also arrive from the focus chain (e.g. tab), which does not
+        # go through self.focus().
         if self.text_input is not None:
             self.text_input.focus()
 

--- a/src/textual_textarea/text_editor.py
+++ b/src/textual_textarea/text_editor.py
@@ -126,30 +126,31 @@ class TextAreaPlus(TextArea, inherit_bindings=False):
         Binding("shift+down", "cursor_down(True)", "cursor down select", show=False),
         Binding("shift+left", "cursor_left(True)", "cursor left select", show=False),
         Binding("shift+right", "cursor_right(True)", "cursor right select", show=False),
-        # Binding("f5", "select_word", "select word", show=False),
-        # Binding("f6", "select_line", "select line", show=False),
         Binding("ctrl+a", "select_all", "select all", show=False),
-        # Editing
+        # Editing. The super+ aliases are the cmd key on macOS; Textual reports
+        # them on terminals that support the Kitty keyboard protocol.
         Binding("ctrl+underscore", "toggle_comment", "toggle comment", show=False),
-        Binding("ctrl+x", "cut", "copy", show=False),
-        Binding("ctrl+c", "copy", "copy", show=False),
-        Binding("ctrl+u,ctrl+v,shift+insert", "paste", "paste", show=False),
-        Binding("ctrl+z", "undo", "undo", show=False),
-        Binding("ctrl+y", "redo", "redo", show=False),
+        Binding("ctrl+x,super+x", "cut", "cut", show=False),
+        Binding("ctrl+c,super+c", "copy", "copy", show=False),
+        Binding("ctrl+u,ctrl+v,super+v,shift+insert", "paste", "paste", show=False),
+        Binding("ctrl+z,super+z", "undo", "undo", show=False),
+        Binding("ctrl+y,super+y", "redo", "redo", show=False),
         # Deletion
         Binding("backspace", "delete_left", "delete left", show=False),
         Binding("delete", "delete_right", "delete right", show=False),
         Binding("shift+delete", "delete_line", "delete line", show=False),
-        # Binding(
-        #     "ctrl+w", "delete_word_left", "delete left to start of word", show=False
-        # ),
-        # Binding(
-        #     "ctrl+f", "delete_word_right", "delete right to start of word", show=False
-        # ),
-        # Binding(
-        #     "ctrl+u", "delete_to_start_of_line", "delete to line start", show=False
-        # ),
-        # Binding("ctrl+k", "delete_to_end_of_line", "delete to line end", show=False),
+        Binding(
+            "ctrl+backspace,alt+backspace",
+            "delete_word_left",
+            "delete left to start of word",
+            show=False,
+        ),
+        Binding(
+            "alt+delete",
+            "delete_word_right",
+            "delete right to start of word",
+            show=False,
+        ),
     ]
 
     clipboard: str = ""
@@ -460,6 +461,10 @@ class TextAreaPlus(TextArea, inherit_bindings=False):
                 self.get_cursor_line_end_location(),
             )
             self.clipboard = f"{whole_line}{self.document.newline}"
+        # Textual's own clipboard: sets App.clipboard and emits OSC 52, which
+        # reaches the system clipboard over ssh and in terminals where pyperclip
+        # has no backend.
+        self.app.copy_to_clipboard(self.clipboard)
         if self.use_system_clipboard and self.system_copy is not None:
             try:
                 self.system_copy(self.clipboard)
@@ -1017,6 +1022,7 @@ class TextEditor(Widget, can_focus=True, can_focus_children=False):
             self.post_message(TextAreaClipboardError(action="copy"))
             return
         self.text_input.clipboard = text
+        self.app.copy_to_clipboard(text)
         if self.use_system_clipboard and self.text_input.system_copy is not None:
             try:
                 self.text_input.system_copy(text)
@@ -1158,9 +1164,8 @@ class TextEditor(Widget, can_focus=True, can_focus_children=False):
 
     @on(TextAreaPlus.Changed)
     def check_for_find_updates(self, event: TextAreaPlus.Changed) -> None:
-        try:
-            find_input = self.footer.query_one(FindInput)
-        except Exception:
+        find_input = self.footer.query_one_optional(FindInput)
+        if find_input is None:
             return
         self._update_find_label(value=find_input.value)
 
@@ -1325,12 +1330,9 @@ class TextEditor(Widget, can_focus=True, can_focus_children=False):
         await self._mount_footer_path_input("open")
 
     async def action_find(self, prepopulate_from_history: bool = False) -> None:
-        try:
-            find_input = self.footer.query_one(FindInput)
-        except Exception:
-            pass
-        else:
-            find_input.focus()
+        existing_input = self.footer.query_one_optional(FindInput)
+        if existing_input is not None:
+            existing_input.focus()
             return
         if prepopulate_from_history and self._find_history:
             value = self._find_history[-1]
@@ -1344,12 +1346,9 @@ class TextEditor(Widget, can_focus=True, can_focus_children=False):
         await self._mount_footer_input(input_widget=find_input)
 
     async def action_goto_line(self) -> None:
-        try:
-            goto_input = self.footer.query_one(GotoLineInput)
-        except Exception:
-            pass
-        else:
-            goto_input.focus()
+        existing_input = self.footer.query_one_optional(GotoLineInput)
+        if existing_input is not None:
+            existing_input.focus()
             return
         goto_input = GotoLineInput(
             max_line_number=self.text_input.document.line_count

--- a/tests/functional_tests/test_autocomplete.py
+++ b/tests/functional_tests/test_autocomplete.py
@@ -133,6 +133,31 @@ async def test_autocomplete(
 
 
 @pytest.mark.asyncio
+async def test_focus_does_not_dismiss_completion_list(
+    app: App, word_completer: Callable[[str], list[tuple[str, str]]]
+) -> None:
+    """
+    Focusing the TextEditor when it already holds focus must not blur the inner
+    TextArea, since the blur dismisses an open completion list.
+    """
+    async with app.run_test() as pilot:
+        ta = app.query_one("#ta", expect_type=TextEditor)
+        ta.word_completer = word_completer
+        assert ta.text_input is not None
+
+        await pilot.press("s")
+        await app.workers.wait_for_complete()
+        await pilot.pause()
+        assert ta.completion_list.is_open is True
+
+        ta.focus()
+        await pilot.pause()
+        assert ta.text_input.has_focus is True
+        assert ta.text_input.completer_active == "word"
+        assert ta.completion_list.is_open is True
+
+
+@pytest.mark.asyncio
 async def test_autocomplete_with_types(
     app: App,
     word_completer_with_types: Callable[[str], list[tuple[tuple[str, str], str]]],

--- a/tests/functional_tests/test_textarea.py
+++ b/tests/functional_tests/test_textarea.py
@@ -246,6 +246,28 @@ from textual_textarea import TextEditor
             "bar\nbaz",
             Selection(start=(0, 0), end=(1, 0)),
         ),
+        # word deletion, using Textual's own delete_word_left/right actions
+        (
+            ["ctrl+backspace"],
+            "foo bar",
+            Selection(start=(0, 7), end=(0, 7)),
+            "foo ",
+            Selection(start=(0, 4), end=(0, 4)),
+        ),
+        (
+            ["alt+backspace"],
+            "foo bar",
+            Selection(start=(0, 7), end=(0, 7)),
+            "foo ",
+            Selection(start=(0, 4), end=(0, 4)),
+        ),
+        (
+            ["alt+delete"],
+            "foo bar",
+            Selection(start=(0, 0), end=(0, 0)),
+            " bar",
+            Selection(start=(0, 0), end=(0, 0)),
+        ),
     ],
 )
 @pytest.mark.asyncio

